src/org/scada_lts/web/mvc/api/EventDetectorAPI.java

```diff
@@ -150,7 +150,12 @@ private JsonPointEventDetector createEventDetector(DataPointVO dataPointVO, Poin
         }
         dataPointVO.getEventDetectors().add(pointEventDetectorVO);
         dataPointService.saveEventDetectors(dataPointVO);
-        Common.ctx.getRuntimeManager().saveDataPoint(dataPointVO);
+        synchronized (eventDetectorsList) {
+            if(!eventDetectorsList.contains(pointEventDetectorVO.getXid())) {
+                Common.ctx.getRuntimeManager().saveDataPoint(dataPointVO);
+                eventDetectorsList.add(pointEventDetectorVO.getXid());
+            }
+        }
```

Comment:
I propose to create a private contains method where I would do this synchronization on the collection, the add method in the synchronized list is already synchronized. Then we should get the same effect (While the thread is executing contains, it will not do remove or add), without double synchronization.